### ISSUE TITLE
About   _C_ops.final_state_dropout  in  paddle.nn.functional.dropout  (interoperability  between  python   and   C++,  CPython  extension,  Python/C  API)

### ISSUE THREAD

**Issue by OccupyMars2030** [PaddlePaddle/Paddle#42012]
### 请提出你的问题 Please ask your question

I have a question  about  the source code of  paddle.nn.functional.dropout  in  the file  python/paddle/nn/functional/common.py.
`        if _non_static_mode():
            if default_main_program().random_seed != 0:
                seed = default_main_program().random_seed

            if in_dygraph_mode():
                out, mask = _C_ops.final_state_dropout( x, None, p, not training, mode, \
                    seed if seed is not None else 0, seed is not None)

                return out
            out, mask = _C_ops.dropout(
                x, 'dropout_prob', p, 'is_test', not training, 'fix_seed',
                seed is not None, 'seed', seed
                if seed is not None else 0, 'dropout_implementation', mode)
            return out
`

In the above code,  I think  _C_ops.dropout   may  be bound to  DropoutOp which is written in  C++,   but  where does  _C_ops.final_state_dropout   come  from ???

**Comment by OccupyMars2030** [comment]
![Screenshot from 2022-04-20 11-14-21](https://user-images.githubusercontent.com/31559413/164143182-6f670484-70d1-4a2e-aa83-370be630ad9a.png)

**Comment by JiabinYang** [comment]
This method in auto_generated during compiling process, which u can find it in `paddle/fluid/pybind/eager_final_state_op_function_impl.h`

**Comment by iclementine** [comment]
The mthod `final_state_dropout` of module `_C_ops` is bound to a c++ function whose signature is:

```c++
static PyObject * eager_final_state_api_dropout(PyObject *self, PyObject *args, PyObject *kwargs)
```

The source of this function and the binding are generated during compilation. The path is of the generated file is `paddle/fluid/pybind/eager_final_state_op_function_impl.h`.

**Comment by OccupyMars2030** [comment]
Thank you for the infomation. Now I know where to learn it.

**Comment by OccupyMars2030** [comment]
In  essence,  it  is  about   interoperability between python and C++.
If  you  want  to  understand  it,   I  recommend  the  following  material  for  you  to  study:


1.Extending and Embedding the Python Interpreter
https://docs.python.org/3.10/extending/index.html
2.Create a C++ extension for Python using Visual  Studio
https://docs.microsoft.com/en-us/visualstudio/python/working-with-c-cpp-python-in-visual-studio?view=vs-2022#debug-the-c-code
https://github.com/microsoft/python-sample-vs-cpp-extension

**Comment by OccupyMars2030** [comment]
<img width="960" alt="image" src="https://user-images.githubusercontent.com/31559413/184542203-2ff1ec28-c484-445c-8536-83f9f574c1f5.png">

**Comment by OccupyMars2030** [comment]
<img width="960" alt="image" src="https://user-images.githubusercontent.com/31559413/184542331-d5d7d40e-a36b-4d84-88af-623e49cc36a3.png">
